src/cargo/core/resolver/mod.rs

```diff
@@ -797,7 +807,10 @@ impl<'a> RegistryQueryer<'a> {
             let b_in_previous = self.try_to_use.contains(b.summary.package_id());
             let a = (a_in_previous, a.summary.version());
             let b = (b_in_previous, b.summary.version());
-            a.cmp(&b).reverse()
+            match self.minimal_versions {
+                true => a.cmp(&b),
+                false => a.cmp(&b).reverse(),
+            }
```

Comment:
This needs your code from https://github.com/rust-lang/cargo/pull/5200#issuecomment-374771936. Also [match on a bool](https://rust-lang-nursery.github.io/rust-clippy/master/index.html#match_bool) is somtimes better as an `if`, and the [`.then_with(`](https://doc.rust-lang.org/stable/std/cmp/enum.Ordering.html#method.then_with) method may be helpfull. I leave it to your judgment.